static/js/controllers/inbox.js

```diff
@@ -362,12 +363,15 @@ var feedback = require('../modules/feedback'),
             return !settings.setup_complete;
           },
           render: function(callback) {
-            $('#guided-setup').modal('show');
-            $('#guided-setup').on('hide.bs.modal', callback);
-            UpdateSettings({ setup_complete: true })
+            $scope.openGuidedSetup()
+              .catch(function() {}) // modal was cancelled - continue
+              .then(function() {
+                return UpdateSettings({ setup_complete: true });
+              })
               .catch(function(err) {
                 $log.error('Error marking setup_complete', err);
-              });
+              })
+              .then(callback);
```

Comment:
can we remove the callback here, and have a promise instead? (for all the `render` funcs)